### DISCUSSION TITLE
'Delete user' feature

### DISCUSSION THREAD

**Discussion by damsleth** [Puzzlepart/did#896] [Feature request]
- [X] I have searched for existing issues to avoid creating duplicates

### TLDR
I want to be able to delete a user from **did**

### Description
With #544 , importing users is super easy.
Deleting users should also be possible in the UI.

This is a **highly destructive action**, something that should be reflected in the UI.

I suggest highlighting that this is a destructive action (red) in the user edit panel, as well as implementing a safety mechanism similar to Github's "delete repository" to prevent accidental deletion.

#### Panel
![image](https://user-images.githubusercontent.com/7300548/93484404-2852fe80-f902-11ea-9838-b7c7e6a09aab.png)

#### Panel button:hover
![image](https://user-images.githubusercontent.com/7300548/93484414-2be68580-f902-11ea-881b-4172bb37868f.png)

### Delete confirmation dialog
![image](https://user-images.githubusercontent.com/7300548/93486219-399d0a80-f904-11ea-9426-fbf7d8f25750.png)

## inspo
![image](https://user-images.githubusercontent.com/7300548/93483875-964af600-f901-11ea-9565-1e27b33d344e.png)

![image](https://user-images.githubusercontent.com/7300548/93483884-9814b980-f901-11ea-8fed-cf17ef8d9c45.png)

## Releated issue
#548

**Comment by okms** [comment]
Do we have a way of disabling a user? E.g. we should disable but not delete tarald

**Comment by olemp** [comment]
> Do we have a way of disabling a user? E.g. we should disable but not delete tarald

Nope, that's an issue! Post it 👍

**Comment by olemp** [comment]
Should this delete all time entried and confirmed periods for the user aswell? @okms  @damsleth

**Comment by damsleth** [comment]
> Should this delete all time entried and confirmed periods for the user aswell? @okms  @damsleth 

Yes. I was thinking it's the GDPR-compliant "delete my data" function.

**Comment by okms** [comment]
We have to check if we have to completely delete the users data or if we can anonymize

**Comment by damsleth** [comment]
Ties inn strongly with #754 
from [norskmakulering.no](https://blogg.norskmakulering.no/sletter-du-e-post-og-personopplysninger-nar-ansatte-slutter#:~:text=Alle%20arbeidsgivere%20behandler%20personopplysninger%2C%20og,behandlet%20for%2C%20og%20dermed%20slettes.) (relevant bits):

_Merk deg at det ikke er all informasjon som skal slettes. Eksempelvis skal økonomiske opplysninger oppbevares i henhold til bokføringsloven og skattelovgivningen. Arkivloven, som først og fremst gjelder offentlige organ, men i tillegg enkelte private arkivbevarende institusjoner, legger også føringer for hva slags opplysninger som skal lagres._

I interpret this as we can get away with keeping non-anonymized time entries in the database for accounting purposes and still be GDPR compliant.
Also if that's the case - meaning we _have_ to keep the data for accounting purposes, this feature only serves one purpose, which is to delete wrongly added A(A)D users from the users' list, e.g. parkeringskortA@puzzlepart.com

**Comment by olemp** [comment]
Converting to discussion until we have the specification ready.

- Delete data? Anonymize data?
- Confirm delete? How?

cc @damsleth @okms
